unison-cli/src/Unison/Codebase/Editor/Propagate.hs

```diff
@@ -600,12 +605,20 @@ applyDeprecations patch =
 -- | Things in the patch are not marked as propagated changes, but every other
 -- definition that is created by the `Edits` which is passed in is marked as
 -- a propagated change.
-applyPropagate :: Applicative m => Patch -> Edits Symbol -> Branch0 m -> Branch0 m
+applyPropagate :: forall m. Applicative m => Patch -> Edits Symbol -> Branch0 m -> Branch0 m
 applyPropagate patch Edits {..} = do
   let termTypes = Map.map (Hashing.typeToReference . snd) newTerms
   -- recursively update names and delete deprecated definitions
-  Branch.stepEverywhere (updateLevel termReplacements typeReplacements termTypes)
+  stepEverywhereButLib (updateLevel termReplacements typeReplacements termTypes)
   where
+    -- Like Branch.stepEverywhere, but don't step the child named "lib"
+    stepEverywhereButLib :: (Branch0 m -> Branch0 m) -> (Branch0 m -> Branch0 m)
+    stepEverywhereButLib f branch =
+      let children =
+            Map.mapWithKey
+              (\name child -> if name == "lib" then child else Branch.step (Branch.stepEverywhere f) child)
```

Comment:
This spot doesn't handle nested libs either, e.g. `x.foo.bar` would be stepped.

Reply:
What's a nested lib?